ui/app/AppLayouts/Chat/panels/UserListPanel.qml

```diff
@@ -75,66 +76,127 @@ Item {
 
         clip: true
 
-        StatusListView {
-            id: userListView
-            objectName: "userListPanel"
-
-            clip: false
+        Loader {
+            id: loadingUsersView
 
             anchors.fill: parent
-            anchors.bottomMargin: Style.current.bigPadding
-            displayMarginEnd: anchors.bottomMargin
 
-            model: SortFilterProxyModel {
-                sourceModel: root.usersModel
+            active: messageStore.loading
+            visible: active
+            sourceComponent: MessagesLoadingView {
+                isUserList: true
+                anchors.fill: parent
+            }
+        }
+
+
+        Repeater {
```

Comment:
Why is this a Repeater? Perhaps I'm missing something but it shouldn't be like that... you're always interested in one channel's user list right?

Reply:
I used the same pattern that was used in ChatColumnView.

What I think it achieves is that it repeats all the Loaders. They are all vaccinate by default. Then the backend decides which to set active, using the `loaderActive` property. This makes sure only the currently selected chat is active and the last 4 as well. 
This is to make sure switching from one channel to another and back and forth is always instant.

I guess using a ListView wouldn't achieve the same goal as it would unload the one no longer active.

I guess the possible downside is maybe the RAM usage since we're instantiating all those Loaders already. I'll do a before and after to see.

Reply:
I have serious doubts about how it works and if it's correct.

`UserListPanel` is instantiated for every community (in a lazy way, when community is clicked first time). And every instance holds as many loaders as many sections we have. So for 5 communities, when all clicked we probably have at least 25 loaders.

Every instance of `UserListPanel` is specific to given community, `activeChatId` once initialized is never changing later. So from all `Loader` in `Repeater`, at most one is enabled. For that reason the `Repeater` seems to be pointless here. The whole job is done by delayed activation via `Timer` giving impression of more responsible UI (but it's only workaround).

The another concern here is that changes in `UserListPanel` break many points of [contributing](https://github.com/status-im/status-desktop/blob/master/CONTRIBUTING.md)
- context properties injected from backend should be hidden behind stores for testability (like `chatCommunitySectionModule`)
- relatively low-level components like `UserListPanel` should just take a simple model and render it, without doing any special logic inside
- ideally they should not depend on stores, what is needed are models

As an example of well-formed UI component with clean api, no dependencies on stores and no business logic inside we could take e.g. [AssetsView](https://github.com/status-im/status-desktop/blob/master/ui/imports/shared/views/AssetsView.qml). That component takes model containing what's is expected to be rendered (~ what we see in figma). Any necessary data transformations are done in an independent component (adaptor). It makes it easy to develop/maintain and measure performance if needed.

Changes introduced to `UserListPanel` also totally break existing storybook page.

I would be happy to help with code structure and guideline compliance.

Reply:
This ticket is probably related (but not directly): https://github.com/status-im/status-desktop/issues/14964, trying to address some fundamental issues here.

Reply:
I'll address the issues with complying to the contributing guide and fix the storybook page.

I think it will require a bigger refactor if we want to get rid of the Repeater though. Especially since the ChatColumnView also uses that pattern

Reply:
@micieslak thanks for the comments. I updated to make `UserListPanel` have no logic inside. It now only receives a `usersModel`. 
I put all the logic in a new component called  `UserListsView`.

The storybook is fixed:
![image](https://github.com/user-attachments/assets/20aaccc3-bf92-44eb-817e-67591c63bb52)

I'll open an issue so that later we can refactor the code to remove the Repeaters and make it lighter

Reply:
@jrainville, regarding

> I think it will require a bigger refactor if we want to get rid of the Repeater though. Especially since the ChatColumnView also uses that pattern

the problem is that the `Repeater` here make no sense here, the whole improvement is done by just delayed loading via loader/timer. I'm not sure how it look like for `ChatColumnView` but here `UserListsView` is instantiated for every community causing that Repeater inside is just pointless (or I'm really missing sth here).